hal/ia32/cpu.c

```diff
@@ -175,21 +155,21 @@ int hal_cpuReschedule(spinlock_t *spinlock, spinlock_ctx_t *scp)
 		"pushl $0;"
 		"pushl %%eax;"
 		"pushl $0;"
-		"movl %4, %%eax;"
+		"movl %1, %%eax;"
 		"call *%%eax;"
 		"cli;"
 		"addl $12, %%esp;"
 		"popl %%esp;"
 
 		"call _interrupts_multilockClear;"
- 		"pushl %%esp;"
+		"pushl %%esp;"
 
 		"jmp interrupts_popContext;"
 
-		"3:;"
+		"1:;"
 		"movl %%eax, %0"
 	: "=g" (err)
-	: "m" (spinlock), "m" (spinlock->lock), "r" (*scp), "g" (threads_schedule)
+	: "g" (threads_schedule)
 	: "eax", "edx", "esp", "cc", "memory");
```

Comment:
```suggestion
	: "eax", "esp", "cc", "memory");
```
`edx` is not used in the `asm` block.

Reply:
It's there due to cdecl calling convention used for `threads_schedule()` - `eax`, `ecx` and `edx` aren't saved by the callee-side and could be modified by `threads_schedule()`.
Added missing `ecx` in clobber list.